### PR TITLE
Feat: Navigation setup

### PR DESCRIPTION
**Pull request by geriux**
- Moved the `babel` configuration to the `package.json`
- Updated some eslint rules (see comments in the PR)
- Added new dependencies:
    - react-redux
    - redux
    - redux-thunk
    - babel-plugin-module-resolver
    - eslint-import-resolver-babel-module

`babel-plugin-module-resolver` is the one we'll use for 'aliases' so we don't have to actual path of the files when importing

`eslint-import-resolver-babel-module` allows eslint to detect these new aliases

- Added main navigation for the app using `NavigationCardStack` 
- Folder structure as previously discussed with @j8seangel

### CODE REVIEW

File: .eslintrc.yml [hunk 1]
```diff
@@ -17,8 +17,11 @@ rules:
   max-len: [1, 140, 2, {"ignoreUrls":true}]
   comma-dangle: [error, never]
   no-console: [error, allow: [warn, error, info]]
+  react/jsx-filename-extension: [1, { "extensions": [".js", ".jsx"] }]
+  react/forbid-prop-types: 0
```
Comment (geriux) [inline .eslintrc.yml]: By default it doesn't allow these proptypes -> `['any', 'array', 'object']` so I disabled it

File: package.json [hunk 2]
```diff
@@ -25,5 +30,24 @@
...
+      [
+        "module-resolver",
+        {
+          "alias": {
+            "app": "./app",
+            "routes": "./app/routes",
+            "components": "./app/components",
+            "containers": "./app/containers",
+            "redux-modules": "./app/redux-modules"
+          }
```
Comment (geriux) [inline package.json]: Here are the aliases

File: .eslintrc.yml (hunk 1 above)
Comment (j8seangel) [inline .eslintrc.yml]: I don't understand why it didn't happen with us in other projects before but thanks !

File: package.json (hunk 2 above)
Comment (j8seangel) [inline package.json]: This is sooooo useful 😍

**Pull request merged**